android/src/main/java/org/ligi/gobandroid_hd/logic/GoGame.kt

```diff
@@ -302,6 +302,47 @@ class GoGame @JvmOverloads constructor(size: Int, handicap: Int = 0) {
             }
         }
 
+    /**
+     * @return the last variations move
+     */
+    fun lastVarMove(): GoMove?
+        {
+            var move = actMove
+            if (move.isFirstMove) return null
+            move = move.parent
+
+            var lastmove:GoMove?  = null
+            var found:Boolean = false
+            for (next_move_variation in move.nextMoveVariations) {
+                if (next_move_variation == actMove) {
+                    return lastmove
+                }
+                lastmove = next_move_variation
+            }
+            return lastmove
```

Comment:
lastmove will be updated in for loop if there are multi-variations ,  and will return null if there is only one variation.